### PR TITLE
Expand Linux test matrix

### PR DESCRIPTION
**Pull request by thejohnfreeman**
This change makes progress on the goal of #4371. It does not replicate the full [matrix](https://github.com/XRPLF/rippled/blob/d794a0f3f161bb30c74881172fc38f763d7d46e8/.github/README.md#continuous-integration) implemented in #3851, but it does replicate the 1.ii section of the Linux matrix. It leverages our "heavy" self-hosted runners and [my Docker image](https://github.com/thejohnfreeman/rippled-docker/) for a rippled build environment, and demonstrates what I believe is a simple, readable pattern for future matrices.

### CODE REVIEW

**Review by shichengripple001 — APPROVED** [review]
looks good to me

**Comment by intelliot** [comment]
notes:
- flaky tests occur on Mac; is there an equivalent of heavy?
  - no. per @shichengripple001 , we may get a Mac mini and use it for our self-hosted runners.
- this currently removes the Mac job. @thejohnfreeman will add a workflow for Mac.
  - even if it is flaky, we would rather have that, than no job at all.

**Comment by drlongle** [comment]
Looks good to me.

**Review by drlongle — APPROVED** [review]

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        uses: actions/cache@v2
+        with:
+            path: ${{ steps.pip-cache.outputs.dir }}
+            key: ${{ runner.os }}-${{ hashFiles('.github/workflows/nix.yml') }}
+      - name: install Conan
+        run: pip install wheel 'conan<2'
+      - name: check environment
+        run: |
+          echo ${PATH} | tr ':' '\n'
+          python --version
```
Comment (legleux) [inline .github/workflows/macos.yml:45]: Throw this info on the workflow summary?

```
PYTHON_VERSION=$(python --version) 
CONAN_VERSION=$(conan --version)
CMAKE_VERSION=$(cmake --version)

echo "Python | Conan | CMake" >> $GITHUB_STEP_SUMMARY
echo ":-:|:-:|:-:" >> $GITHUB_STEP_SUMMARY
echo "${PYTHON_VERSION} | ${CONAN_VERSION} | ${CMAKE_VERSION}" >> $GITHUB_STEP_SUMMARY
```


`env >> $GITHUB_STEP_SUMMARY` :question:

Comment (thejohnfreeman) [reply .github/workflows/macos.yml:45]: These can be multiline outputs (CMake is), and I don't feel like working around that. If it becomes something we check often and want to make easier to see, then I'll reconsider. Until then, I think what we have now is fine.

File: .github/workflows/nix.yml [hunk 2]
```diff
@@ -3,94 +3,112 @@ on: [push, pull_request]
...
+              version: 11
+              cc: /usr/bin/gcc
+              cxx: /usr/bin/g++
+          - compiler: clang
+            profile:
+              version: 14
+              cc: /usr/bin/clang-14
+              cxx: /usr/bin/clang++-14
+    runs-on: [self-hosted, heavy]
+    container: thejohnfreeman/rippled-build-ubuntu:12e19cd9034b
```
[the remark this answers is not in the document]
Comment (legleux) [reply .github/workflows/nix.yml:30]: So we can test it into perpetuity?

[the remark this answers is not in the document]
Comment (thejohnfreeman) [reply .github/workflows/nix.yml:30]: Ok. I think there's a case to be made to pick one configuration to test on multiple versions of Ubuntu. I don't have an 18.04 image with everything available right now. Won't take me long to make one, but perhaps we can punt that to a future PR?

**Pull request merged**